src/output/scanline_output_file.rs

```diff
@@ -134,10 +159,73 @@ impl<'a> ScanlineOutputFile<'a> {
             let msg = unsafe { CStr::from_ptr(error_out) };
             Err(Error::Generic(msg.to_string_lossy().into_owned()))
         } else {
+            self.scanlines_written = self.header().data_dimensions().1;
```

Comment:
> I don't mean "write extra pixel data" so much as "overwrite the file in-place."

Right, but what I'm saying is the C++ API doesn't do the latter, it does the former.  So we definitely do want to prevent it from being called again.

> If upstream does not permit such a use, is there a good reason not to express that by making this consume self?

Not a strong reason, but I feel like it's more consistent with the incremental write API, as well as the eventual tile writing API which is inherently incremental.  Kind of a, "If we can't generally enforce this statically, let's not try, so that people's expectations stay the same."

But I don't feel strongly about it, and I'm happy to have it consume self if you'd prefer it that way.